r/R/python.R

```diff
@@ -64,3 +64,28 @@ r_to_py.RecordBatch <- function(x, convert = FALSE) {
   ExportRecordBatch(x, array_ptr, schema_ptr)
   pa$RecordBatch$`_import_from_c`(array_ptr, schema_ptr)
 }
+
+#' Install pyarrow for use with reticulate
+#'
+#' `pyarrow` is the Python package for Apache Arrow. This function helps with
+#' installing it for use with `reticulate`.
+#'
+#' @param envname The name or full path of the Python environment to install
+#' into. This can be a virtualenv or conda environment created by `reticulate`.
+#' See `reticulate::py_install()`.
+#' @param nightly: logical: Should we install a development version of the
+#' package? Default is to use the official release version.
+#' @param ... additional arguments passed to `reticulate::py_install()`.
+#' @export
+install_pyarrow <- function(envname = NULL, nightly = FALSE, ...) {
+  if (nightly) {
+    reticulate::py_install("pyarrow", envname = envname, ...,
+      # Nightly for pip
+      pip_options = "--extra-index-url https://repo.fury.io/arrow-nightlies/ --pre --force-reinstall",
```

Comment:
What is `--force-reinstall` for?

Reply:
It was my experience that if I have pyarrow 0.16.0 from pypi installed, and then I tried to install the nightly, it would say "requirements already met" and not install the nightly. Is that... inaccurate? Is there a better flag to use?

Reply:
`--upgrade` probably conveys your intent better then :-)

Reply:
(it also works when the package isn't already installed)